shap/datasets.py

```diff
@@ -259,8 +314,7 @@ def cache(url, file_name=None):
     if file_name is None:
         file_name = os.path.basename(url)
     data_dir = os.path.join(os.path.dirname(__file__), "cached_data")
-    if not os.path.isdir(data_dir):
-        os.mkdir(data_dir)
+    os.makedirs(data_dir, exist_ok=True)
```

Comment:
`pathlib` would be nice... but that is out of scope of this PR :)

Reply:
Agreed, when we have time, I want to remove all py2 legacy code (still plenty lying around the codebase), and start using some of the newer language features.